mne/decoding/base.py

```diff
@@ -672,3 +648,266 @@ def get_coef(estimator, attr='filters_', inverse_transform=False):
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+          - None, to use the default 3-fold cross validation,
+          - integer, to specify the number of folds in a `(Stratified)KFold`,
+          - An object to be used as a cross-validation generator.
+          - An iterable yielding train, test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass, :class:`StratifiedKFold` is used. In all
+        other cases, :class:`KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    n_jobs : integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose : integer, optional
+        The verbosity level.
+
+    fit_params : dict, optional
+        Parameters to pass to the fit method of the estimator.
+
+    pre_dispatch : int, or string, optional
+        Controls the number of jobs that get dispatched during parallel
+        execution. Reducing this number can be useful to avoid an
+        explosion of memory consumption when more jobs get dispatched
+        than CPUs can process. This parameter can be:
+
+            - None, in which case all the jobs are immediately
+              created and spawned. Use this for lightweight and
+              fast-running jobs, to avoid delays due to on-demand
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    Returns
+    -------
+    scores : array of float, shape=(len(list(cv)),) | array of array
+        Array of scores of the estimator for each run of the cross validation.
+
+    """
+    # This code is copied from sklearn
+
+    from sklearn.base import is_classifier, clone
+    from sklearn.utils import indexable
+    from sklearn.metrics.scorer import check_scoring
+    from sklearn.model_selection._split import check_cv
+
+    X, y, groups = indexable(X, y, groups)
+
+    cv = check_cv(cv, y, classifier=is_classifier(estimator))
+    cv_iter = list(cv.split(X, y, groups))
+    scorer = check_scoring(estimator, scoring=scoring)
+    # We clone the estimator to make sure that all the folds are
+    # independent, and that it is pickle-able.
+    # Note: this parallelization is implemented using MNE Parallel
+    parallel, p_func, n_jobs = parallel_func(_fit_and_score, n_jobs,
+                                             pre_dispatch=pre_dispatch)
+    scores = parallel(p_func(clone(estimator), X, y, scorer, train, test,
+                             verbose, None, fit_params)
+                      for train, test in cv_iter)
+    return np.array(scores)[:, 0, ...]
```

Comment:
maybe it would be nicer to use a common Python expression for the flattening problem

```scores = sum(scores, list())```

or 

```scores = [a for b in scores for a in b]```